### PR TITLE
Persist first location

### LINKED ISSUE #761: Record location just before showing the timeline
After the onboarding, if user enabled location, we'll need to get the user's current location before loading the timeline.

When the user enabled location not from the onboarding dialog, but from the custom alert in the timeline, we also need to get the current location.

A loading indicator is needed as this will take 2-3 seconds.

### PR DESCRIPTION
**Pull request by Odrakir**
Closes #761

### CODE REVIEW

File: teferi/UI/Modules/Main/MainViewModel.swift [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (sofokli1) [inline teferi/UI/Modules/Main/MainViewModel.swift]: interesting place to add a new line but ok

File: teferi/UI/Common/Views/LoadingView.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+class LoadingView: UIView
+{
+    static let sharedInstance = LoadingView()
+    
+    private let backgroundView = UIView()
+    private let loadingIndicator = LoadingIndicator()
+    private let label = UILabel()
+    
+    var text: NSAttributedString? = nil {
```
Comment (sofokli1) [inline teferi/UI/Common/Views/LoadingView.swift]: new line

File: teferi/UI/Common/Views/LoadingView.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        didSet {
```
Comment (sofokli1) [inline teferi/UI/Common/Views/LoadingView.swift]: new line

File: teferi.xcworkspace/contents.xcworkspacedata [hunk 4]
```diff
@@ -2,6 +2,9 @@
 <Workspace
    version = "1.0">
    <FileRef
+      location = "group:/Users/Ricardo/Documents/Proyectos/Toggl/Superday/teferi/UI/Common/Views/LoadingIndicator.swift">
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply teferi.xcworkspace/contents.xcworkspacedata]: 🤷‍♂️ This was the file giving me problems in bitrise. Probably I deleted something I shouldn't have when merging... It's ok now.

File: teferi/UI/Common/Views/LoadingView.swift (hunk 2 above)
Comment (Odrakir) [reply teferi/UI/Common/Views/LoadingView.swift]: You mean the "{" should go in a new line? I thought we agree to enforce it only for methods and classes (structs, enums and extensions too). That's how the linter is configured.

File: teferi/UI/Modules/Main/MainViewModel.swift (hunk 1 above)
Comment (Odrakir) [reply teferi/UI/Modules/Main/MainViewModel.swift]: Deleted.

File: teferi/UI/Common/Views/LoadingView.swift (hunk 3 above)
Comment (Odrakir) [reply teferi/UI/Common/Views/LoadingView.swift]: Same as above.

File: teferi.xcworkspace/contents.xcworkspacedata (hunk 4 above)
[the remark this answers is not in the document]
Comment (sofokli1) [reply teferi.xcworkspace/contents.xcworkspacedata]: is ti ok that it says `/Users/Ricardo/Documents/Proyectos...`?

[the remark this answers is not in the document]
Comment (Odrakir) [reply teferi.xcworkspace/contents.xcworkspacedata]: Fuck... not, that's not ok.

**Pull request merged**